### PR TITLE
[CI] Add `--debug` flag to the `linter.go` invoke tasks to better debug golangci-lint witchcraft

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR adds a `--debug` flag to the `linter.go` invoke task.

### Motivation

 Better debugging of golangci-lint lint discrepancies due to version mismatches.

### CODE REVIEW

**Review by CelianR — APPROVED** [review]

File: tasks/libs/common/check_tools_version.py [hunk 1]
```diff
@@ -45,27 +48,41 @@ def custom_golangci_v(v: str) -> str:
...
     Check that each installed tool in tools_list is the version expected for the repo.
     """
-    is_expected_versions = True
+    should_exit = False
     tools_versions = {
-        'go': {'current_v': current_go_v(ctx), 'expected_v': expected_go_repo_v()},
+        'go': {
+            'current_v': current_go_v(ctx),
+            'expected_v': expected_go_repo_v(),
+            'debug': current_go_v(ctx),
```
Comment (KevinFairise2) [inline tasks/libs/common/check_tools_version.py:70]: Should we make it empty if `--debug` is not passed? To have the same behavior between `golangci-lint` and `go`

**Review by KevinFairise2 — APPROVED** [review]
LGTM.

I feel like we could always compute the `debug` and print it in the logs instead of adding a new flag that we enable everywhere. Because the debug information are lightweight and should not make too much noise

File: tasks/libs/common/check_tools_version.py [hunk 2]
```diff
@@ -45,27 +48,41 @@ def custom_golangci_v(v: str) -> str:
...
+            'exit_on_error': False,
+            'error_msg': "Warning: If you have linter errors it might be due to version mismatches.",
+        },
         'golangci-lint': {
             'current_v': current_golangci_lint_v(ctx),
             'expected_v': custom_golangci_v(expected_golangci_lint_repo_v(ctx)),
+            'debug': '' if not debug else current_golangci_lint_v(ctx, debug=debug),
+            'exit_on_error': True,
+            'error_msg': "Error: The golanci-lint version you are using is not the correct one. Please run inv -e install-tools to install the correct version.",
```
Comment (pducolin) [inline tasks/libs/common/check_tools_version.py:79]: **💬 suggestion**
Would suggest running `inv -e setup` here to install any missing tooling at once

File: tasks/libs/common/check_tools_version.py [hunk 3]
```diff
@@ -74,9 +91,14 @@ def check_tools_version(ctx: Context, tools_list: list[str]) -> bool:
         else:
             current_v, expected_v = tools_versions[tool]['current_v'], tools_versions[tool]['expected_v']
             if current_v != expected_v:
-                is_expected_versions = False
                 print(
-                    f"Warning: Expecting {tool} '{expected_v}' but you have {tool} '{current_v}'. Please fix this as you might encounter issues using the tooling.",
+                    color_message(
+                        f"Expecting {tool} '{expected_v}' but you have {tool} '{current_v}'. Please fix this as you might encounter issues using the tooling.",
```
Comment (pducolin) [inline tasks/libs/common/check_tools_version.py:96]: **💬 suggestion**
Should you suggest also gere to run `inv -e setup`?

**Review by pducolin — APPROVED** [review]

File: tasks/libs/common/check_tools_version.py (hunk 1 above)
Comment (amenasria) [reply tasks/libs/common/check_tools_version.py]: Good point, fixed with [7139880](https://github.com/DataDog/datadog-agent/pull/29802/commits/7139880450e426b6600152bb98c6d4c181af4f2a).

File: tasks/libs/common/check_tools_version.py (hunk 2 above)
Comment (amenasria) [reply tasks/libs/common/check_tools_version.py]: Good point, fixed with [d9f11cb](https://github.com/DataDog/datadog-agent/pull/29802/commits/d9f11cb15a85c4171e55cb6d6d4c6a33e01c9799).

File: tasks/libs/common/check_tools_version.py (hunk 3 above)
Comment (amenasria) [reply tasks/libs/common/check_tools_version.py]: Good point, fixed with [d9f11cb](https://github.com/DataDog/datadog-agent/pull/29802/commits/d9f11cb15a85c4171e55cb6d6d4c6a33e01c9799).

**Comment by amenasria** [comment]
@KevinFairise2 
> I feel like we could always compute the debug and print it in the logs instead of adding a new flag that we enable everywhere. Because the debug information are lightweight and should not make too much noise

I agree that debug could be the default behavior if they weren't producing a lot of logs. That's sadly not the case for `golangci-lint`, using the `--debug` flag actually adds ~200 lines in the logs, hence the section and the feature.

**Review by clarkb7 — APPROVED** [review]

**Pull request merged**